server/routes/urlRoutes.js

```diff
@@ -1,8 +1,9 @@
 const express = require("express");
-const { getSpecificUrl, createUrl } = require("../controller/urlController");
+const { getSpecificUrl, createUrl, getAllUserUrls } = require("../controller/urlController");
 
 const urlRouter = new express.Router();
 
+urlRouter.get("/v1/apis/urls", getAllUserUrls);
```

Comment:
create a middleware `file named authMiddleware inside middlewares folder` to check user is authenticated or not.

Then add middleware here like: 
`urlRouter.get("/v1/apis/urls", isAuthenticated, getAllUserUrls);`

see reference: https://github.com/abhijeetnishal/social-media-app/blob/master/server/src/middlewares/auth.ts

Reply:
Hey @atharvarekhawar, i have added reference how do we implement middlewares in routes intead of controller.

Reply:
but from where will we get the userId of user if it is not passed in req @abhijeetnishal

Reply:
> but from where will we get the userId of user if it is not passed in req @abhijeetnishal

You can get id from routes like this:
`postRouter.get("/following-users/:id", isAuthenticated, followingUsersPosts);`

and in controller use req.params

Reply:
but how will you do it for urlRouter.get("/v1/apis/urls", isAuthenticated, getAllUserUrls);

Reply:
It will be urlRouter.get("/v1/apis/urls/:id", isAuthenticated, getAllUserUrls);

Reply:
but /urls/:id makes more sense , when you want to fetch a specific URL rather than passing userId 

your thoughts?